### ISSUE TITLE
Allow alternative config file naming like Nlog does.

### ISSUE THREAD

**Issue by Muhomorik** [Microsoft/ApplicationInsights-dotnet-logging#222]
In the Classic desktop work it is common to deploy multiple applications in the same folder.

Project structure is then:

- Project
-- WPF app1
-- WPF app2

The installer (like WIX) then takes the result of the build and places in the same folder, creates start menu items and shortcuts.
Usually, app1 and app2 are always deployed together and create only one entry in the "Add/remove" programs.

This is not a problem for NLog because:

> The following locations will be searched when executing a stand-alone *.exe application:
> 
>     - standard application configuration file (usually applicationname.exe.config)
>     - applicationname.exe.nlog in application’s directory
>     - NLog.config in application’s directory (Name sensitive; using docker dotnet core)
>     - NLog.dll.nlog in a directory where NLog.dll is located (only if NLog isn't installed in the GAC)

Basically, WPF_app1.exe.config solves the problem when two exes are deployed together with specific loggers for each one of those.

For the development purposes I have the following structure:

- ApplicationInsights.config
- ApplicationInsights.Deployment.config
- ApplicationInsights.Test.config

VS uses the default and sends to debug listener only, during test, data is send to the separate instance (who want to see breakpoints) and when deployer, installer takes the deploy config.

Nothing new :)

However, when deploying two apps together there could be only one ApplicationInsights.config file for both apps.

**Comment by MS-TimothyMothra** [comment]
Hi @Muhomorik, thanks for the interest in our SDK :)

Respectfully, this is not a feature that we're likely to support.

You may instead be interested in an in-code implementation of the config. I understand that may be less than ideal, but that should help achieve your scenario.

I'm going to close this issue for now, but if you have more questions feel free to re-open. 
Thanks,
